### PR TITLE
3411 define events

### PR DESCRIPTION
**Pull request by mssalvatore**
# What does this PR do?

Adds events that will be used by the cryptojacker.
Issue #3411

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Do all unit tests pass?
* [ ] Do all end-to-end tests pass?
* [ ] Any other testing performed?
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

**Review by ilija-lazoroski — APPROVED** [review]

File: monkey/common/agent_events/http_request_event.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from http import HTTPMethod
+
+from pydantic import AnyHttpUrl
+
+from . import AbstractAgentEvent
+
+
+class HTTPRequestEvent(AbstractAgentEvent):
+    """
+    An event that occurs when the Agent consumes significant RAM for its own purposes.
```
Comment (cakekoa) [inline monkey/common/agent_events/http_request_event.py:10]: Is this related to RAM?

**Review by cakekoa — APPROVED** [review]
